src/Microsoft.DotNet.Internal.DependencyInjection.Testing/DependencyInjectionValidation.cs

```diff
@@ -59,28 +64,28 @@ public static class DependencyInjectionValidation
                 continue;
             }
 
-            if (IsExemptType(serviceImplementationType) || IsExemptType(service.ServiceType))
+            if (IsExemptType(serviceImplementationType, exemptTypes) || IsExemptType(service.ServiceType, exemptTypes))
```

Comment:
Can't we just go at the beginning of the `foreach` and do "if exempt type, continue"?